programs/lib/startup.c

```diff
@@ -34,6 +34,7 @@ int main(void);
 /**
  * Allocate a requested memory and return a pointer to it.
  */
+#ifndef BOOTLOADER
```

Comment:
rather than conditional compilation, perhaps a separate `startup_bootloader.c`?

Reply:
Sure we can factor it out into a separate startup file and allow the startup file to be set from a Makefile and then not be overriden by the Makefrag.